osu.Game/Rulesets/UI/RulesetContainer.cs

```diff
@@ -43,7 +42,7 @@ public abstract class RulesetContainer : Container
         /// <summary>
         /// The input manager for this RulesetContainer.
         /// </summary>
-        internal readonly PlayerInputManager InputManager = new PlayerInputManager();
+        internal IHasReplayHandler ReplayInputManager => (IHasReplayHandler)KeyBindingInputManager;

```

Comment:
Then you need to check the usings of RulesetContainer.HasReplayLoaded, because that will certainly nullref, even if a replay hasn't been attempted to be loaded.